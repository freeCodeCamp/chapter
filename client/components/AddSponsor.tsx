import React, { useState } from 'react';
import { useForm, Controller } from 'react-hook-form';
import {
  Button,
  TextField,
  makeStyles,
  Select,
  MenuItem,
  InputLabel,
  FormControl,
} from '@material-ui/core';

const useStyles = makeStyles(() => ({
  form: {
    display: 'flex',
    flexDirection: 'column',
    maxWidth: '25%',
  },
  item: {
    marginTop: '20px',
  },
  responseDiv: {
    margin: '15px 0',
  },
}));

const AddSponsor: React.FC = () => {
  const [responseMsg, setResponseMsg] = useState('');
  const styles = useStyles();

  const { control, handleSubmit } = useForm();

  // TODO: Get data from store
  // const eventId = useSelector(state => state.selectedChapter.eventId);
  // const chapterId = useSelector(state => state.selectedChapter.id);

  const onSubmit = async data => {
    try {
      // await dispatch(sponsorActions.submit(eventId, chapterId));
      setResponseMsg(`${data.name} has been added as a ${data.type} sponsor.`);
    } catch (e) {
      setResponseMsg('Uh oh, something went wrong.');
      // TODO: more descriptive error messages
    }
  };

  return (
    <>
      {responseMsg && <div className={styles.responseDiv}>{responseMsg}</div>}
      <form onSubmit={handleSubmit(onSubmit)} className={styles.form}>
        <FormControl className={styles.item}>
          <Controller
            control={control}
            as={
              <TextField
                name="name"
                type="text"
                label="Sponsor Name"
                placeholder="Glitter and Sparkle Co"
              />
            }
            name="name"
            options={{ required: true }}
          />
        </FormControl>
        <FormControl className={styles.item}>
          <Controller
            control={control}
            as={
              <TextField
                label="Sponsor Website"
                name="website"
                type="text"
                placeholder="www.glitter.co"
              />
            }
            name="website"
            options={{ required: true }}
          />
        </FormControl>
        <FormControl className={styles.item}>
          <InputLabel id="sponsor-type-label">Sponsor Type</InputLabel>
          <Controller
            control={control}
            as={
              <Select labelId="sponsor-type-label">
                <MenuItem value={'FOOD'}>Food</MenuItem>
                <MenuItem value={'BEVERAGE'}>Beverage</MenuItem>
                <MenuItem value={'OTHER'}>Other</MenuItem>
              </Select>
            }
            name="type"
            defaultValue={'OTHER'}
            options={{ required: true }}
          />
        </FormControl>
        <Button
          className={styles.item}
          variant="contained"
          color="primary"
          type="submit"
        >
          Add Sponsor
        </Button>
      </form>
    </>
  );
};

export default AddSponsor;
