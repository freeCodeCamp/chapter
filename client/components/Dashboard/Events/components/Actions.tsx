import React from 'react';
import { Button } from '@material-ui/core';
import Link from 'next/link';

import { eventActions } from 'client/store/actions';
import useConfirm from 'client/hooks/useConfirm';
import useThunkDispatch from 'client/hooks/useThunkDispatch';
import { IEventModal } from 'client/store/types/events';

interface IActionsProps {
  event: IEventModal;
  onDelete?: () => void;
}

const Actions: React.FC<IActionsProps> = ({ event, onDelete }) => {
  const dispatch = useThunkDispatch();

  const [confirmCancel, clickCancel] = useConfirm(() =>
    dispatch(eventActions.cancelEvent(1, event.id)),
  );
  const [confirmRemove, clickRemove] = useConfirm(() => {
    dispatch(eventActions.removeEvent(1, event.id));
    onDelete && onDelete();
  });

  return (
    <>
      {!event.canceled && (
        <Button onClick={clickCancel}>
          {confirmCancel ? 'Are you sure?' : 'Cancel'}
        </Button>
      )}
      <Button onClick={clickRemove}>
        {confirmRemove ? 'Are you sure?' : 'delete'}
      </Button>
      <Link
        href={`/dashboard/events/[id]/edit`}
        as={`/dashboard/events/${event.id}/edit`}
      >
        <a style={{ marginLeft: '10px' }}>Edit</a>
      </Link>
    </>
  );
};

export default Actions;
