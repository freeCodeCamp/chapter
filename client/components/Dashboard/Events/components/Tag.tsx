import React from 'react';
import { makeStyles } from '@material-ui/core';

import { ITagModal } from 'client/store/types/events';

const useStyles = makeStyles({
  tag: {
    borderRadius: '8px',
    padding: '4px 12px',
    backgroundColor: '#22c98c',
    margin: '8px 2px',
    color: '#fff',
    fontWeight: 600,
  },
  tagsContainer: { display: 'flex' },
});

const Tags: React.FC<{ tags?: ITagModal[] }> = ({ tags }) => {
  const style = useStyles();

  return tags ? (
    <div className={style.tagsContainer}>
      {tags.map(tag => (
        <div className={style.tag} key={`tag-${tag.id}`}>
          {tag.name}
        </div>
      ))}
    </div>
  ) : null;
};

export default Tags;
