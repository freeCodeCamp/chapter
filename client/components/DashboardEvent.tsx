import React from 'react';
import { Card, Typography } from '@material-ui/core';

import ProgressCardContent from './ProgressCardContent';
import { IEventModal } from 'client/store/types/events';
import Link from 'next/link';

interface IDashboardEventProps {
  event: IEventModal;
  loading: boolean;
}

const DashboardEvent: React.FC<IDashboardEventProps> = ({ event, loading }) => {
  return (
    <Card style={{ marginTop: '12px' }}>
      <ProgressCardContent loading={loading}>
        <Link
          href="/dashboard/events/[id]"
          as={`/dashboard/events/${event.id}`}
        >
          <a>
            <Typography gutterBottom variant="h5" component="h2">
              {event.name}
            </Typography>
          </a>
        </Link>
        <Typography variant="body2" color="textSecondary" component="p">
          {event.description}
        </Typography>
        <Typography variant="body2" color="textSecondary" component="p">
          {event.capacity}
        </Typography>
      </ProgressCardContent>
    </Card>
  );
};

export default DashboardEvent;
