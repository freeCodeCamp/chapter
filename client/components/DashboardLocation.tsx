import React, { useState } from 'react';
import {
  Card,
  Typography,
  Button,
  makeStyles,
  Link as MULink,
} from '@material-ui/core';
import { useDispatch } from 'react-redux';
import Link from 'next/link';

import ProgressCardContent from './ProgressCardContent';
import { ILocationModal } from 'client/store/types/locations';
import { locationActions } from 'client/store/actions';

interface IDashboardLocationProps {
  location: ILocationModal;
  loading: boolean;
}

const useStyles = makeStyles(() => ({
  bottom: {
    display: 'flex',
    alignItems: 'center',
  },
  action: {
    marginLeft: '20px',
  },
}));

const DashboardLocation: React.FC<IDashboardLocationProps> = ({
  location,
  loading,
}) => {
  const [confirm, setConfirm] = useState<boolean>(false);
  const dispatch = useDispatch();

  const styles = useStyles();

  const remove = () => {
    dispatch(locationActions.remove(location.id));
  };

  return (
    <Card style={{ marginTop: '12px' }}>
      <ProgressCardContent loading={loading}>
        <MULink gutterBottom variant="h5" component="h2">
          <Link href={`/dashboard/locations/${location.id}`}>
            {location.city}
          </Link>
        </MULink>
        <Typography variant="body2" color="textSecondary" component="p">
          {`${location.region}, ${location.country_code}, ${location.postal_code}`}
        </Typography>
        <Typography variant="body2" color="textSecondary" component="p">
          {location.address}
        </Typography>

        <div className={styles.bottom}>
          {confirm ? (
            <Button onClick={remove}>Are you sure</Button>
          ) : (
            <Button onClick={() => setConfirm(true)}>DELETE</Button>
          )}

          <Link href={`/dashboard/locations/${location.id}/edit`}>
            <a className={styles.action}>Edit</a>
          </Link>
        </div>
      </ProgressCardContent>
    </Card>
  );
};

export default DashboardLocation;
