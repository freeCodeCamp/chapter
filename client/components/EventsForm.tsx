import React from 'react';
import { useForm, Controller } from 'react-hook-form';
import { FormControl, TextField, Button, makeStyles } from '@material-ui/core';

interface IField {
  key: string;
  label: string;
  placeholder?: string;
  type: string;
  defaultValue?: string;
}

const fields: IField[] = [
  {
    key: 'name',
    type: 'text',
    label: 'Event title',
    placeholder: 'Foo and the Bars',
  },
  {
    key: 'description',
    type: 'text',
    label: 'Description',
    placeholder: '',
  },
  {
    key: 'capacity',
    type: 'number',
    label: 'Capacity',
    placeholder: '50',
  },
  {
    key: 'tags',
    type: 'text',
    label: 'Tags (separated by a comma)',
    placeholder: 'Foo, bar',
  },
  {
    key: 'start_at',
    type: 'datetime-local',
    label: 'Start at',
    defaultValue: new Date().toISOString().slice(0, 16),
  },
  {
    key: 'end_at',
    type: 'datetime-local',
    label: 'End at',
    defaultValue: new Date(Date.now() + 1000 * 60 * 60)
      .toISOString()
      .slice(0, 16),
  },
];

const useStyles = makeStyles(() => ({
  form: {
    display: 'flex',
    flexDirection: 'column',
    maxWidth: '25%',
  },
  item: {
    marginTop: '20px',
  },
}));

export interface IEventFormData {
  name: string;
  description: string;
  capacity: number;
  tags: string;
}

interface IEventsFormProps {
  onSubmit: (data: IEventFormData) => void;
}

const EventsForm: React.FC<IEventsFormProps> = ({ onSubmit }) => {
  const { control, handleSubmit } = useForm();
  const styles = useStyles();

  return (
    <form onSubmit={handleSubmit(onSubmit)} className={styles.form}>
      {fields.map(field => (
        <FormControl className={styles.item} key={field.key}>
          <Controller
            control={control}
            as={
              <TextField
                label={field.label}
                type={field.type}
                placeholder={field.placeholder}
              />
            }
            name={field.key}
            defaultValue={field.defaultValue}
            options={{ required: true }}
          />
        </FormControl>
      ))}
      <Button
        className={styles.item}
        variant="contained"
        color="primary"
        type="submit"
      >
        Add Event
      </Button>
    </form>
  );
};

export default EventsForm;
