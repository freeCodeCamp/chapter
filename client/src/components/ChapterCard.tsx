import {
  Grid,
  Text,
  GridItem,
  Flex,
  Box,
  Image,
  Tooltip,
} from '@chakra-ui/react';
import { Link } from 'chakra-next-link';
import React from 'react';

import { isPast } from 'date-fns';
import { LockIcon } from '@chakra-ui/icons';
import { ChaptersQuery } from '../generated/graphql';

type ChapterCardProps = {
  chapter: ChaptersQuery['chapters'][number];
};

export const ChapterCard: React.FC<ChapterCardProps> = ({ chapter }) => {
  return (
    <>
      {chapter.logo_url && (
        <Image
          src={chapter.logo_url}
          alt={`${chapter.name} logo`}
          position="relative"
          top="2.75em"
          left="2em"
          width="4.5em"
          height="4.5em"
          borderRadius="50%"
          outline="2px solid white"
        />
      )}
      <Box
        data-cy="chapter-card"
        borderWidth="1px"
        borderRadius="lg"
        overflow="hidden"
        width={'full'}
        minW={'20em'}
        gap={'2'}
        backgroundImage={
          chapter.banner_url ??
          'https://cdn.freecodecamp.org/chapter/orange-graphics-small.jpg'
        }
        backgroundPosition={'center'}
        backgroundRepeat={'no-repeat'}
        backgroundSize={'cover'}
      >
        <Grid
          templateColumns="repeat(3, 1fr)"
          gap={'3'}
          width="100%"
          marginRight={'1em'}
          paddingBlock={'.5em'}
          color={'gray.00'}
          bgImage="linear-gradient(to bottom,hsl(240 14% 27%/ .9),  hsl(240 14% 10%/ .9), hsl(240 14% 10%))"
          alignItems="center"
          templateAreas={`
          ". . ."
          ". . ."
          "chaptername chaptername subnumber"
          "aboutheader aboutheader aboutheader"
          "about about about"
          "eventheader eventheader eventheader"
          "event event event"
          "event event event"
          `}
        >
          <GridItem
            paddingInline={'1em'}
            paddingBlock={'.5em'}
            area="chaptername"
          >
            <Link
              fontSize={['lg', 'xl', '2xl']}
              fontWeight={700}
              fontFamily={'body'}
              paddingInline=".1em"
              href={`/chapters/${chapter.id}`}
              data-cy="chaptercard-name"
            >
              {chapter.name}
            </Link>
          </GridItem>
          <GridItem
            display="inline-grid"
            width="100%"
            justifyItems="flex-end"
            gridArea="subnumber"
            paddingInline="1.5em"
          >
            <Text fontWeight="bold">
              Members: {chapter._count.chapter_users}
            </Text>
          </GridItem>
          <Text
            paddingInline={'1em'}
            fontSize={['md', 'lg', 'xl']}
            fontWeight={'500'}
            gridArea="aboutheader"
          >
            About
          </Text>
          <Text
            noOfLines={3}
            paddingInline={'1em'}
            fontWeight={400}
            fontSize={['sm', 'md', 'lg']}
            gridArea="about"
          >
            {chapter.description}
          </Text>
          <Text
            fontSize={['md', 'lg', 'xl']}
            fontWeight={'500'}
            paddingInline={'1em'}
            marginBlockStart={'.5em'}
            gridArea="eventheader"
          >
            Next Events
          </Text>
          <GridItem area="event" paddingInline={'1em'}>
            {chapter.events
              .filter(({ ends_at }) => !isPast(new Date(ends_at)))
              .map(({ id, name, start_at, invite_only }) => (
                <Flex
                  paddingBlock={'.5em'}
                  paddingInline={'.3em'}
                  justifyContent={'space-between'}
                  alignItems="center"
                  key={id}
                >
                  <Link
                    href={`/events/${id}`}
                    mt="2"
                    fontWeight={600}
                    fontSize={['sm', 'md', 'lg']}
                  >
                    {name}
                  </Link>
                  {invite_only && (
                    <Tooltip label="Invite only">
                      <LockIcon
                        mt="2"
                        marginLeft="auto"
                        marginRight="1"
                        fontSize={['sm', 'md', 'lg']}
                      />
                    </Tooltip>
                  )}
                  <Text mt="2" fontWeight={600} fontSize={['sm', 'md', 'lg']}>
                    {isPast(new Date(start_at)) ? 'Running' : 'Upcoming'}
                  </Text>
                </Flex>
              ))}
          </GridItem>
        </Grid>
      </Box>
    </>
  );
};
