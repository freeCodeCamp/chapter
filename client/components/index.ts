import SomeComponent from './SomeComponent';
import AddSponsor from './AddSponsor';
import ProgressCardContent from './ProgressCardContent';

export { SomeComponent, AddSponsor, ProgressCardContent };
