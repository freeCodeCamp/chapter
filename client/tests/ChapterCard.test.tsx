/**
 * @jest-environment jsdom
 */

import { render } from '@testing-library/react';
import React from 'react';
import { ChakraProvider } from '@chakra-ui/react';
import { ChapterCard } from '../src/components/ChapterCard';

describe('ChapterCard', () => {
  it('should render', () => {
    jest.useFakeTimers().setSystemTime(new Date('2023-01-18'));
    const { container } = render(
      <ChakraProvider>
        <ChapterCard
          chapter={{
            name: 'Name of the Chapter',
            description: 'Schaefer, McLaughlin and Conn',
            id: 1,
            banner_url: 'https://loremflickr.com/640/480/tech?75964',
            logo_url: 'https://loremflickr.com/150/150/tech?30938',
            events: [
              {
                canceled: true,
                id: 1,
                name: 'Canceled event',
                start_at: new Date('2023-01-20 10:00'),
                ends_at: new Date('2023-01-20 10:30'),
                invite_only: true,
              },
              {
                canceled: false,
                id: 6,
                name: 'Next event',
                start_at: new Date('2023-01-21 10:00'),
                ends_at: new Date('2023-01-21 10:30'),
                invite_only: false,
              },
              {
                canceled: false,
                id: 3,
                name: 'Past event',
                start_at: new Date('2023-01-15 20:00'),
                ends_at: new Date('2023-01-15 20:30'),
                invite_only: false,
              },
              {
                canceled: false,
                id: 5,
                name: 'Visible event',
                start_at: new Date('2023-01-25 08:00'),
                ends_at: new Date('2023-01-26 08:00'),
                invite_only: false,
              },
            ],
            _count: { chapter_users: 3 },
          }}
        />
      </ChakraProvider>,
    );

    expect(container).toMatchSnapshot();
  });
});
