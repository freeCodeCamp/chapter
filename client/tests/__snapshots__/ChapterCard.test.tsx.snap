// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChapterCard should render 1`] = `
<div>
  <img
    alt="Name of the Chapter logo"
    class="chakra-image css-mjr311"
    src="https://loremflickr.com/150/150/tech?30938"
  />
  <div
    class="css-r2nu6e"
    data-cy="chapter-card"
  >
    <div
      class="css-1fcrsi5"
    >
      <div
        class="css-1dhwyht"
      >
        <a
          class="chakra-link css-1100460"
          data-cy="chaptercard-name"
          href="/chapters/1"
        >
          Name of the Chapter
        </a>
      </div>
      <div
        class="css-7yu0ub"
      >
        <p
          class="chakra-text css-722v25"
        >
          Members: 
          3
        </p>
      </div>
      <p
        class="chakra-text css-1pn1wzz"
      >
        About
      </p>
      <p
        class="chakra-text css-1k7k16j"
      >
        Schaefer, McLaughlin and Conn
      </p>
      <p
        class="chakra-text css-1sccb2s"
      >
        Next Events
      </p>
      <div
        class="css-60517v"
      >
        <div
          class="css-1flr4u6"
        >
          <a
            class="chakra-link css-lps4mj"
            href="/events/1"
          >
            Canceled event
          </a>
          <svg
            class="chakra-icon css-3ymgfo"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19.5,9.5h-.75V6.75a6.75,6.75,0,0,0-13.5,0V9.5H4.5a2,2,0,0,0-2,2V22a2,2,0,0,0,2,2h15a2,2,0,0,0,2-2V11.5A2,2,0,0,0,19.5,9.5Zm-9.5,6a2,2,0,1,1,3,1.723V19.5a1,1,0,0,1-2,0V17.223A1.994,1.994,0,0,1,10,15.5ZM7.75,6.75a4.25,4.25,0,0,1,8.5,0V9a.5.5,0,0,1-.5.5H8.25a.5.5,0,0,1-.5-.5Z"
              fill="currentColor"
            />
          </svg>
          <p
            class="chakra-text css-dn4mnl"
          >
            Upcoming
          </p>
        </div>
        <div
          class="css-1flr4u6"
        >
          <a
            class="chakra-link css-lps4mj"
            href="/events/6"
          >
            Next event
          </a>
          <p
            class="chakra-text css-dn4mnl"
          >
            Upcoming
          </p>
        </div>
        <div
          class="css-1flr4u6"
        >
          <a
            class="chakra-link css-lps4mj"
            href="/events/5"
          >
            Visible event
          </a>
          <p
            class="chakra-text css-dn4mnl"
          >
            Upcoming
          </p>
        </div>
      </div>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
